test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -61,10 +61,30 @@ def create_mom_hook(self):
                 'alarm': '3', 'enabled': 'true'}
         self.server.create_import_hook(name, attr, body)
 
+    def create_bad_begin_hook(self):
+        name = "h2"
+        body = ("import pbs\n"
+                "e=pbs.event()\n"
+                "if e.job.in_ms_mom():\n"
+                "    e.accept()\n"
+                "x\n")
+        attr = {'event': 'execjob_begin', 'fail_action': 'offline_vnodes'}
+        self.server.create_import_hook(name, attr, body)
+
+    def create_bad_startup_hook(self):
+        name = "h3"
+        body = ("import pbs\n"
+                "x\n")
+        attr = {'event': 'exechost_startup', 'fail_action': 'offline_vnodes'}
+        self.server.create_import_hook(name, attr, body)
+
     def create_multi_vnodes(self, num_moms, num_vnode=3):
-        if num_moms != len(self.moms.values()):
+        try:
             self.server.manager(MGR_CMD_DELETE, NODE, id="@default",
                                 expect=True)
+        except:
+            pass

```

Comment:
This is the wrong thing to do.  If you fail to delete all your nodes, your system will be in an unstable state for the rest of the test.  You should let manager() throw its exception.